tests/Reminder/Reminder.test.ts

```diff
@@ -0,0 +1,22 @@
+/**
+ * @jest-environment jsdom
+ */
+jest.mock('obsidian');
+import moment from 'moment';
+import { Reminder, ReminderType, parseMoment } from '../../src/Reminders/Reminder';
+
+window.moment = moment;
+
+describe('Reminder ', () => {
+    it('testing parsing Moment() datetime', () => {
+        const reminder = moment('2023-04-30 11:44 am');
+
+        expect(parseMoment(reminder)).toStrictEqual(new Reminder(moment('2023-04-30 11:44 am'), ReminderType.DateTime));
+    });
+
+    it('testing parsing Moment() date', () => {
+        const reminder = moment('2023-04-30');
+
+        expect(parseMoment(reminder)).toStrictEqual(new Reminder(moment('2023-04-30'), ReminderType.Date));
```

Comment:
Running this test gives me this console output:

```
/usr/local/bin/node --require /Users/clare/Library/Application Support/JetBrains/Toolbox/apps/WebStorm/ch-0/223.8617.44/WebStorm.app/Contents/plugins/javascript-impl/helpers/jest-intellij/lib/jest-intellij-stdin-fix.js /Users/clare/Documents/develop/Obsidian/schemar/obsidian-tasks/node_modules/jest/bin/jest.js --colors --reporters /Users/clare/Library/Application Support/JetBrains/Toolbox/apps/WebStorm/ch-0/223.8617.44/WebStorm.app/Contents/plugins/javascript-impl/helpers/jest-intellij/lib/jest-intellij-reporter.js --verbose --runTestsByPath /Users/clare/Documents/develop/Obsidian/schemar/obsidian-tasks/tests/Reminder/Reminder.test.ts
  console.warn
    Deprecation warning: value provided is not in a recognized RFC2822 or ISO format. moment construction falls back to js Date(), which is not reliable across all browsers and versions. Non RFC2822/ISO date formats are discouraged. Please refer to http://momentjs.com/guides/#/warnings/js-date/ for more info.
    Arguments: 
    [0] _isAMomentObject: true, _isUTC: false, _useUTC: false, _l: undefined, _i: 2023-04-30 11:44 am, _f: undefined, _strict: undefined, _locale: [object Object]
    Error: 
        at Function.createFromInputFallback (/Users/clare/Documents/develop/Obsidian/schemar/obsidian-tasks/node_modules/moment/moment.js:324:25)
        at configFromString (/Users/clare/Documents/develop/Obsidian/schemar/obsidian-tasks/node_modules/moment/moment.js:2550:19)
        at configFromInput (/Users/clare/Documents/develop/Obsidian/schemar/obsidian-tasks/node_modules/moment/moment.js:2993:13)
        at prepareConfig (/Users/clare/Documents/develop/Obsidian/schemar/obsidian-tasks/node_modules/moment/moment.js:2976:13)
        at createFromConfig (/Users/clare/Documents/develop/Obsidian/schemar/obsidian-tasks/node_modules/moment/moment.js:2943:44)
        at createLocalOrUTC (/Users/clare/Documents/develop/Obsidian/schemar/obsidian-tasks/node_modules/moment/moment.js:3037:16)
        at createLocal (/Users/clare/Documents/develop/Obsidian/schemar/obsidian-tasks/node_modules/moment/moment.js:3041:16)
        at hooks (/Users/clare/Documents/develop/Obsidian/schemar/obsidian-tasks/node_modules/moment/moment.js:16:29)
        at Object.<anonymous> (/Users/clare/Documents/develop/Obsidian/schemar/obsidian-tasks/tests/Reminder/Reminder.test.ts:14:32)
        at Promise.then.completed (/Users/clare/Documents/develop/Obsidian/schemar/obsidian-tasks/node_modules/jest-circus/build/utils.js:289:28)
        at new Promise (<anonymous>)
        at callAsyncCircusFn (/Users/clare/Documents/develop/Obsidian/schemar/obsidian-tasks/node_modules/jest-circus/build/utils.js:222:10)
        at _callCircusTest (/Users/clare/Documents/develop/Obsidian/schemar/obsidian-tasks/node_modules/jest-circus/build/run.js:248:40)
        at processTicksAndRejections (node:internal/process/task_queues:96:5)
        at _runTest (/Users/clare/Documents/develop/Obsidian/schemar/obsidian-tasks/node_modules/jest-circus/build/run.js:184:3)
        at _runTestsForDescribeBlock (/Users/clare/Documents/develop/Obsidian/schemar/obsidian-tasks/node_modules/jest-circus/build/run.js:86:9)
        at _runTestsForDescribeBlock (/Users/clare/Documents/develop/Obsidian/schemar/obsidian-tasks/node_modules/jest-circus/build/run.js:81:9)
        at run (/Users/clare/Documents/develop/Obsidian/schemar/obsidian-tasks/node_modules/jest-circus/build/run.js:26:3)
        at runAndTransformResultsToJestFormat (/Users/clare/Documents/develop/Obsidian/schemar/obsidian-tasks/node_modules/jest-circus/build/legacy-code-todo-rewrite/jestAdapterInit.js:120:21)
        at jestAdapter (/Users/clare/Documents/develop/Obsidian/schemar/obsidian-tasks/node_modules/jest-circus/build/legacy-code-todo-rewrite/jestAdapter.js:79:19)
        at runTestInternal (/Users/clare/Documents/develop/Obsidian/schemar/obsidian-tasks/node_modules/jest-runner/build/runTest.js:367:16)
        at runTest (/Users/clare/Documents/develop/Obsidian/schemar/obsidian-tasks/node_modules/jest-runner/build/runTest.js:444:34)

      12 | describe('Reminder ', () => {
      13 |     it('testing parsing Moment() datetime', () => {
    > 14 |         const reminder = moment('2023-04-30 11:44 am');
         |                                ^
      15 |
      16 |         expect(parseMoment(reminder)).toStrictEqual(new Reminder(moment('2023-04-30 11:44 am'), ReminderType.DateTime));
      17 |     });

      at warn (node_modules/moment/moment.js:287:21)
      at Function.createFromInputFallback (node_modules/moment/moment.js:319:17)
      at configFromString (node_modules/moment/moment.js:2550:19)
      at configFromInput (node_modules/moment/moment.js:2993:13)
      at prepareConfig (node_modules/moment/moment.js:2976:13)
      at createFromConfig (node_modules/moment/moment.js:2943:44)
      at createLocalOrUTC (node_modules/moment/moment.js:3037:16)
      at createLocal (node_modules/moment/moment.js:3041:16)
      at hooks (node_modules/moment/moment.js:16:29)
      at Object.<anonymous> (tests/Reminder/Reminder.test.ts:14:32)

```

Reply:
Nice catch. Added the format string. Not sure if it will work, but just notices that moment() can accept multiple format strings and pick the best one. Might allow me to simplify some parsing logic, but would probably still need the 12/24 selector for when serializing reminder dates. https://github.com/users/Erik-Handeland/projects/2/views/1?pane=issue&itemId=27509756